OpenRA.Mods.Common/Scripting/LuaScript.cs

```diff
@@ -19,34 +19,37 @@
 namespace OpenRA.Mods.Common.Scripting
 {
 	[Desc("Part of the new Lua API.")]
-	public class LuaScriptInfo : ITraitInfo, Requires<SpawnMapActorsInfo>
+	public class LuaScriptInfo : ConditionalTraitInfo, ITraitInfo, Requires<SpawnMapActorsInfo>
 	{
 		public readonly HashSet<string> Scripts = new HashSet<string>();
 
-		public object Create(ActorInitializer init) { return new LuaScript(this); }
+		public override object Create(ActorInitializer init) { return new LuaScript(init.Self, this); }
 	}
 
-	public class LuaScript : ITick, IWorldLoaded, INotifyActorDisposing
+	public class LuaScript : ConditionalTrait<LuaScriptInfo>, ITick, IWorldLoaded, INotifyActorDisposing
 	{
 		readonly LuaScriptInfo info;
 		ScriptContext context;
 		bool disposed;
 
-		public LuaScript(LuaScriptInfo info)
+		public LuaScript(Actor self, LuaScriptInfo info)
+			: base(info)
 		{
 			this.info = info;
 		}
 
 		void IWorldLoaded.WorldLoaded(World world, WorldRenderer worldRenderer)
 		{
-			var scripts = info.Scripts ?? Enumerable.Empty<string>();
+			var scripts = IsTraitDisabled ? Enumerable.Empty<string>() : info.Scripts ?? Enumerable.Empty<string>();
```

Comment:
What happens / what do we want to happen if the trait is enabled after the game starts?

IMO the best solution here is to override `TraitEnabled` and throw an exception if the trait is enabled after the game has started.  The victory condition traits should do the same thing, otherwise e.g. `StrategicProgressWidget` will need to account for the victory trait changing.